src/python/WMComponent/WorkflowUpdater/SiteListPoller.py

```diff
@@ -0,0 +1,152 @@
+        :param config: a Configuration object with the component configuration
+        """
+        BaseWorkerThread.__init__(self)
+        myThread = threading.currentThread()
+        self.logger = myThread.logger
+
+        # get wmstats parameters
+        self.wmstatsUrl = getattr(config.WorkflowUpdater, "wmstatsUrl")
+        self.wmstatsSrv = WMStatsServer(self.wmstatsUrl, logger=self.logger)
+        self.states = getattr(config.WorkflowUpdater, "states", ['running-open', 'acquired'])
+
+        # provide access to WMBS in local WMAgent
+        self.daoFactory = DAOFactory(package="WMCore.WMBS",
+                                     logger=myThread.logger,
+                                     dbinterface=myThread.dbi)
+        # DB function to retrieve active workflows
+        self.listActiveWflows = self.daoFactory(classname="Workflow.GetUnfinishedWorkflows")
+
+        # local WorkQueue service
+        self.localCouchUrl = config.WorkQueueManager.couchurl
+        self.localWQ = WorkQueue(self.localCouchUrl,
+                                 config.WorkQueueManager.dbname)
+
+    def getActiveWorkflows(self):
+        """
+        Provide list of active requests within WMAgent
+        :return: dict of workflows names vs pickle files
+        """
+        # get list of active workflows in WMAgent
+        wflowSpecs = self.listActiveWflows.execute()
+
+        # construct dictionary of workflow names and their pickle files
+        wmaSpecs = {}
+        for wflowSpec in wflowSpecs:
+            name = wflowSpec['name']  # this is the name of workflow
+            pklFileName = wflowSpec['spec']  # the "spec" in WMBS table (wmbs_workflow.spec) is pkl file name
+            wmaSpecs[name] = pklFileName
+        return wmaSpecs
+
+    def getRequestSites(self, requests):
+        """
+        Return list of requests specs from WMStats for provided list of request names
+        :param requests: list of workflow requests names
+        :return: dict of workflow records obtained from wmstats server:
+        {"wflow": {"SiteWhitelist":[], "SiteBlacklist": []}, ...}
+        """
+        # get list of workflows from wmstats
+        outputMask = ['SiteWhiteList', 'SiteBlackList']
+        wdict = {}
+        for state in self.states:
+            inputConditions = {"RequestStatus": state}
+            self.logger.info("Fetch site info from WMStats for condition: %s and mask %s", inputConditions, outputMask)
+            resp = self.wmstatsSrv.getFilteredActiveData(inputConditions, outputMask)
+            data = json.loads(resp['data'])
+            for rdict in data['result']:
+                # rdict here has the following structure:
+                # {"RequestName": "bla", "SiteWhitelist":[], "SiteBlacklist": []}
+                wflow = rdict.pop('RequestName')
+                # check that our workflow is in our requests list
+                if wflow in requests:
+                    wdict[wflow] = rdict
```

Comment:
The data coming from this URL: https://cmsweb.cern.ch/wmstatsserver/data/filtered_requests?RequestStatus=running-open&mask=SiteWhitelist&mask=SiteBlacklist has this structure:
```
{"result": [
 {
  "SiteWhitelist": [
    "T1_US_FNAL"
  ],
  "SiteBlacklist": [],
  "RequestName": "cmsunified_ACDC1_Run2024G_Muon1_ZMu_PromptMUODPGNano_241105_094516_3263"
},{
  "SiteWhitelist": [
    "T1_US_FNAL"
  ],
  "SiteBlacklist": [],
  "RequestName": "cmsunified_ACDC1_Run2024G_Muon1_ZMu_PromptMUODPGNano_241105_094536_6961"
},....
```
And, the code has appropriate comment about it. The memory impact is very hard to measure as Python allocate memory based on amount of data it may need to hold. But I agree that dict assigned to dict may have negligible allocation. due to referencing.